### PR TITLE
Fix cpplint CI job and fix all the style issues for that

### PR DESCRIPTION
**Pull request by rcurtin**
I noticed that the style checks job actually wasn't working because the cpplint version isn't working anymore.  I updated the jenkins-conf repo to use an updated version, and then went through all of the various issues it pointed out.

I don't think it captures everything in our style guide, but it gets a lot of stuff, so it's at least an improvement.

Hopefully I should iterate a few times here and then have a fully working style guide build.

### CODE REVIEW

**Comment by zoq** [comment]
@mlpack-jenkins test this please

**Comment by rcurtin** [comment]
@mlpack-jenkins test this please

**Comment by rcurtin** [comment]
This is ready now, but the documentation build will fail until some #3783 is merged (I didn't fix that here).

**Review by zoq — APPROVED** [review]
Wow, so many style fixes.

File: src/mlpack/bindings/markdown/print_docs.cpp [hunk 1]
```diff
@@ -256,7 +256,8 @@ void PrintDocs(const string& bindingName,
     }
 
     cout << "```" << BindingInfo::Language() << endl;
-    cout << example.substr(0, example.size() - 1) << endl; // do not want the last "\n".
+    cout << example.substr(0, example.size() - 1) << endl;
+    // do not want the last "\n".
```
Comment (shrit) [inline src/mlpack/bindings/markdown/print_docs.cpp:260]: It is fine to have that, but I suppose the comment needs to be above as the linter moved it to the next line

File: src/mlpack/bindings/python/is_serializable.hpp [hunk 2]
```diff
@@ -43,8 +43,8 @@ void IsSerializable(util::ParamData& data,
       IsSerializable<typename std::remove_pointer<T>::type>(data);
 }
 
-} // python
-} // bindings
-} // mlpack
+} // namespace python
+} // namespace bindings
+} // namespace mlpack
```
Comment (shrit) [inline src/mlpack/bindings/python/is_serializable.hpp:49]: Are we okay with removing the namespace word?

**Review by shrit — APPROVED** [review]
Looks good I went over half of them, and everything looks good,

File: src/mlpack/bindings/python/is_serializable.hpp (hunk 2 above)
Comment (shrit) [reply src/mlpack/bindings/python/is_serializable.hpp:49]: Ignore my above comment, I was looking at the old code, it seems that the linter has added it, I was surprised that we did not have the namespace word, the code was probably too old. :+1:

File: src/mlpack/bindings/markdown/print_docs.cpp (hunk 1 above)
Comment (rcurtin) [reply src/mlpack/bindings/markdown/print_docs.cpp]: Good point, fixed in 4e14b782. :+1:

**Pull request merged**